processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -995,6 +996,27 @@ public CursorOffsetHolderValueMatcherFactory(
     @Override
     public ValueMatcher makeValueMatcher(String dimension, final Comparable value)
     {
+      if (dimension.equals(Column.TIME_COLUMN_NAME)) {
```

Comment:
this block of code seems copy and pasted, can we put it on a common location somewhere?